### DISCUSSION TITLE
How to handle playlist in custom extractor?

### DISCUSSION THREAD

**Discussion by Lutra-Fs** [Androz2091/discord-player#1759] [Q&A] [answered]
Greetings! I am the creator of Flowease and I require assistance regarding the proper handling of playlists. I have observed that the official extractor has linked various types such as songs, playlists, and albums to the source. However, third-party extractors do not have access to these classes. My current strategy involves generating a distinctive string and comparing it to ascertain if the request is for a playlist. However, I am open to adopting a type-based solution for easier maintenance. Thank you!

**Comment by aka-pal** [comment]
it has
```js
import { BaseExtractor, Playlist,Track } from "discord-player"
```
[Check doc](https://discord-player.js.org/docs/discord-player?type=class&target=Playlist)

**Reply by Lutra-Fs** [reply]
That is in the extractor. I would like to know how to identify if a user would select a playlist instead of a song.

**Reply by aka-pal** [reply]
Use some RegEx to check if it's a playlist url or track url.

**Reply by twlite** [reply, marked as answer]
It's up to you on what to return from your extractor. `QueryType` constant is for discord-player's internal implementation of extractors, which has been around for quite a while. At the moment, you cannot extend `QueryType` values so you have to implement the validator yourself. However, you are able to target a specific extractor to execute by using `ext:EXTRACTOR_ID` as `searchEngine` value.
